### PR TITLE
Fix integer overflow in subvector() function

### PR DESCRIPTION
**Pull request by hlinnaka**
`end = start + count` can overflow if `start` is very large. That leads to a segfault later in the function. Add test case for it.

### CODE REVIEW

File: src/halfvec.c [hunk 1]
```diff
@@ -934,17 +934,32 @@ halfvec_subvector(PG_FUNCTION_ARGS)
...
+	else
+		end = start + count;
 
 	/* Indexing starts at 1, like substring */
 	if (start < 1)
 		start = 1;
-
-	if (end > a->dim)
-		end = a->dim + 1;
+	else if (start > a->dim)
```
Comment (ankane) [inline src/halfvec.c:959]: I think we can let this fall through to `CheckDim` in both functions.

**Review by ankane — COMMENTED** [review]
Thanks @hlinnaka, another great catch. Added a comment inline.

**Comment by hlinnaka** [comment]
Another approach would be to change `end` and `dim` into int64. That might be less error-prone. I think you'd need to change `CheckDim` to also take an int64 argument, but I guess that's fine too.

**Comment by ankane** [comment]
Let's go with this approach for now. Will add a few more test cases to make sure everything is covered.

Thanks again!

**Pull request merged**